src/librustc_session/parse.rs

```diff
@@ -174,6 +174,10 @@ impl ParseSess {
         &self.source_map
     }
 
+    pub fn clone_source_map(&self) -> Lrc<SourceMap> {
```

Comment:
Hm, should we avoid the clone_ here since it's really cheap to clone it (since it's just an Arc)?

Alternatively maybe the previously existing method can be made to return `&Lrc<SourceMap>` and then we could clone that in callers.

Reply:
This is only used once from rustdoc, source map doesn't generally need cloning.
Perhaps we'll be able to replace `Lrc<SourceMap>` with just `SourceMap` in the future.
So I'd rather keep this method explicit / visible for now.